### PR TITLE
feat: add force volume unpublish

### PR DESCRIPTION
**Pull request by tiagolobocastro**
When a node is not reachable we cannot unpublish.
Add a force option which will ignore the error if the node is not online.
Should be used with care, only when the caller knows that it is safe to do so.

### CODE REVIEW

File: control-plane/agents/core/src/volume/specs.rs [hunk 1]
```diff
@@ -592,8 +592,30 @@ impl ResourceSpecsLocked {
...
+                        let node_online = match registry.get_node_wrapper(&nexus_clone.node).await {
+                            Ok(node) => {
+                                let mut node = node.lock().await;
+                                node.is_online() && node.liveness_probe().await.is_ok()
+                            }
+                            _ => false,
+                        };
+                        if !node_online {
+                            nexus_clone.warn_span(|| {
+                                tracing::warn!("Force unpublish. Forgetting about the target nexus because the node is not online and it was requested")
```
Comment (paulyoong) [inline control-plane/agents/core/src/volume/specs.rs:611]: Could we include the nexus ID and node?

File: control-plane/agents/core/src/volume/tests.rs [hunk 2]
```diff
@@ -861,6 +856,26 @@ async fn publishing_test(cluster: &Cluster) {
...
+
+    UnpublishVolume::new(&volume_state.uuid, false)
+        .request()
+        .await
+        .expect_err("The node is not online...");
+
+    UnpublishVolume::new(&volume_state.uuid, true)
+        .request()
+        .await
+        .expect("With force comes great responsibility...");
```
Comment (paulyoong) [inline control-plane/agents/core/src/volume/tests.rs:870]: :grin: May the force be with you!

**Review by paulyoong — APPROVED** [review]

File: control-plane/agents/core/src/volume/specs.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/agents/core/src/volume/specs.rs:611]: It's already automatically included by the nexus :)

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**